### ISSUE TITLE
Alternative interoperable output format to stack dot

### ISSUE THREAD

**Issue by mboes** [commercialhaskell/stack#4837]
Stack can tell you what the dependency graph for the transitive closure of all your packages looks like, using the `stack dot` command, but only in GraphViz format. That's fine for the purposes of visualization by the end user, but brittle for other machines to parse.

We could have say `stack ls dependencies --graph`, that lists the dependencies together with their direct dependencies, i.e. an adjacency list. Either something like

```
conduit: base bytestring directory exceptions ...
primitive: base ghc-prim transformers
```

or as JSON output:

```
{ "conduit": ["base", "bytestring", "directory", "exceptions", ...], "primitive": [...]}
```

cc @aherrmann

**Comment by qrilka** [comment]
@mboes  I suppose #4424 won't solve this issue as it's supposed to output flat dependency list?

**Comment by mboes** [comment]
If #4424 ends up including the dependencies of each item in the list, then that should work fine yeah.

**Comment by akshaymankar** [comment]
Hi, I was almost done with #4424, when I saw this so I decided not to add more things in it.

I think this should be trivial to add to `ls dependencies json`. I'd be happy to implement this. 

I propose  adding `dependencies` field to each package listed in the JSON. This field will have an array with just the names of all dependencies of the package as the issue specifies. Let me know if this works or if there are better ideas.
